stumpy/ostinato.py

```diff
@@ -58,10 +58,9 @@ def _across_series_nearest_neighbors(
     nns_subseq_idx = np.zeros(k, dtype=np.int64)
 
     for i in range(k):
-        QT = core.sliding_dot_product(Ts[Ts_idx][subseq_idx : subseq_idx + m], Ts[i])
-        distance_profile = core._mass(
-            Q,
-            Ts[i],
+        QT = core.sliding_dot_product(Q, Ts[i])
```

Comment:
Since `Q` and `T` are UNPROCESSED, `QT[idx]` can get meaningless value for a non-finite subsequence that starts at `idx`. Better to just replace them with `np.nan`. So, for example, we can do something like this after computing `QT` (`QT = core.sliding_dot_product(Q, Ts[i])`):

```
# post-processing QT
if np.any(~np.isfinite(Q)):
     QT[:] = np.nan

QT[~np.isfinite(M_Ts[i])] = np.nan
``` 


Note that `QT[idx]` will be ignored though eventually once it comes to computing its corresponding value in  the distance profile via `core.calculate_distance_profile` (see a few lines below).  In other words, we do not need the "post-processing of QT" but I think it makes code clearer.

Any concerns?

Reply:
I understand the need to replace the ugly `Ts[Ts_idx][subseq_idx : subseq_idx + m]` with `Q`. That's fine but

```
QT = core.sliding_dot_product(Q, Ts[i])
```

doesn't seem to require any post-processing when either `Q` or `T` contain `np.nan`.

Reply:
May bad! You are right. No need for post-processing `QT`

Reply:
@seanlaw 
> maybe I should better stop here and try to address https://github.com/TDAmeritrade/stumpy/issues/1011 first. Then come back and resume the work on this PR.

I did an experiment and shared the result in https://github.com/TDAmeritrade/stumpy/issues/1011. We can either work on that and then come back here. Or, on the second thought, we can still merge this (after I resolve the merge conflicts), and then work on https://github.com/TDAmeritrade/stumpy/issues/1011.

What do you think?

Reply:
Since we have a merge conflict here anyways, let's fix #1011 first and then come back here and strikethrough the comment:

> Note: Since we pass inputs with non-finite values to core.calculate_distance_profile (Not only QT but also M_T), maybe I should better stop here and try to address https://github.com/TDAmeritrade/stumpy/issues/1011 first. Then come back and resume the work on this PR.

so that we are focused on this issue/PR instead of being distracted by jumping back and forth. I am getting confused by having so many things open and so it feels disorganized